src/rpc/interfaces.h

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_RPC_INTERFACES_H
+#define BITCOIN_RPC_INTERFACES_H
+
+#include <atomic>
+#include <functional>
+#include <list>
+#include <map>
+#include <string>
+#include <vector>
+
+#include <rpc/request.h>
+#include <univalue.h>
+#include <sync.h>
+
+static const unsigned int DEFAULT_RPC_SERIALIZE_VERSION = 1;
+
+static std::atomic<bool> fRequestShutdown(false);
```

Comment:
Making this static and in a .h file means that every compilation unit (cpp file) that includes it will get its own copy of the variable. If you only want a single copy, it must be declared in a .cpp file, and defined in the corresponding .h file.

I also think it's very strange that shutdown is now merged into an RPC-specific module. Does nothing outside of RPC need it?

Reply:
> Making this static and in a .h file means that every compilation unit (cpp file) that includes it will get its own copy of the variable. If you only want a single copy, it must be declared in a .cpp file, and defined in the corresponding .h file.
> 

Thanks for the hint. Will correct it now.

> I also think it's very strange that shutdown is now merged into an RPC-specific module. Does nothing outside of RPC need it?

I never wanted to remove shutdown.h/cpp, but as I could never get pass the linker error I spoke above, the only way for me was to include shutdown's four methods into rpc/interface.

I haven't encountered any problems with shutdown removed. Functional RPC-tests are passing.  

But I'll gladly revert it, if there is a way to avoid the linker error.

Reply:
> it must be declared in a .cpp file, and defined in the corresponding .h file.
> 

I must admit that I don't know how to do this. Especially with const variables and atomics.

Reply:
I'm being stupid sorry - it's a static variable that's local to shutdown.cpp. It can't and doesn't need to go in shutdown.h.

I think you shouldn't need to touch this at all. It's perfectly fine as-as in shutdown.cpp. It also has no dependencies, so touching it shouldn't be needed for the purpose of breaking circular dependencies.

The issue you're seeing while linking must have to do with the organization into libraries and possibly the order in which those are passed to the linker. That should be solvable by potentially moving things between libraries, but it's hard to say what the exact issue is without tinkering with it myself.

Reply:
> I'm being stupid sorry - it's a static variable that's local to shutdown.cpp. It can't and doesn't need to go in shutdown.h.

Thanks for the clarification! I was already questioning life and existence. :)

> 
> I think you shouldn't need to touch this at all. It's perfectly fine as-as in shutdown.cpp. It also has no dependencies, so touching it shouldn't be needed for the purpose of breaking circular dependencies.
> 
> The issue you're seeing while linking must have to do with the organization into libraries and possibly the order in which those are passed to the linker. That should be solvable by potentially moving things between libraries, but it's hard to say what the exact issue is without tinkering with it myself.

Ok, then I will bring back shutdown.h/cpp and then try to find a way.